File: src/main/kotlin/sigma/software/leovegas/drugstore/controller/ProductController.java
package sigma.software.leovegas.drugstore.controller;

import java.util.List;
import lombok.AllArgsConstructor;
import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.*;
import sigma.software.leovegas.drugstore.dto.ProductDto;
import sigma.software.leovegas.drugstore.service.ProductServiceI;

@RestController
@RequestMapping("/products")
@AllArgsConstructor
public class ProductController {
    private final ProductServiceI service;

    @GetMapping
    @ResponseStatus(HttpStatus.OK)
    public List<ProductDto> getProducts(){
        return service.getAll();
    }

    @GetMapping("/{id}")
    @ResponseStatus(HttpStatus.OK)
    public ProductDto getOne(@PathVariable("id") Long id){
        return service.getOne(id);
    }

    @PostMapping
    @ResponseStatus(HttpStatus.CREATED)
    private ProductDto add(@RequestBody ProductDto productDto){
        return service.create(productDto);
    }

    @PutMapping("/{id}")
    @ResponseStatus(HttpStatus.CREATED)
    private ProductDto update(@PathVariable Long id, @RequestBody ProductDto productDto){
        return service.update(id, productDto);
    }

    @DeleteMapping("/{id}")
    @ResponseStatus(HttpStatus.NO_CONTENT)
    private void update(@PathVariable Long id){
        service.delete(id);
    }
}

File: src/main/kotlin/sigma/software/leovegas/drugstore/controller/ProductController.kt
package sigma.software.leovegas.drugstore.controller;

import org.springframework.web.bind.annotation.DeleteMapping
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.PutMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;
import sigma.software.leovegas.drugstore.dto.ProductRequest
import sigma.software.leovegas.drugstore.dto.ProductResponse
import sigma.software.leovegas.drugstore.service.ProductServiceI

@RestController
@RequestMapping("/products")
class ProductController(val service: ProductServiceI) {

    @GetMapping
    fun getAll(): MutableList<ProductResponse> {
        return service.getAll()
    }

    @GetMapping("/{id}")
    fun getOne(@PathVariable id: Long): ProductResponse {
        return service.getOne(id)
    }

    @PostMapping
    fun save(@RequestBody product: ProductRequest): ProductResponse {
        return service.save(product)
    }

    @PutMapping("/{id}")
    fun update(@PathVariable id: Long, @RequestBody product: ProductRequest): ProductResponse {
        return service.update(id, product)
    }

    @DeleteMapping("/{id}")
    fun delete(@PathVariable id: Long){
        service.delete(id)
    }
}
